### DISCUSSION TITLE
Uptime CI not running every 5 minutes

### DISCUSSION THREAD

**Discussion by rouilj** [upptime/upptime#112] [Q&A] [answered]
I set up an upptime instance at https://github.com/rouilj/RoundupAssets.

If I look at the Actions listing and select the Uptime CI, I don't see it being executed every 5 minutes. It looks like it's 18 or so minutes.

If I look at this instance https://github.com/upptime/upptime/actions?query=workflow%3A%22Uptime+CI%22, it looks
like the interval is not 5 minutes.

Any idea on how to troubleshoot this problem or get more info about when the uptime ci instance is run?

Thanks.

-- rouilj

**Comment by AnandChowdhary** [comment, marked as answer]
Hi @rouilj, unfortunately this is out of our control. We request GitHub to run the Uptime CI workflow every 5 minutes. It's added to a queue and run whenever GitHub has a machine available, sometimes this is every 7 minutes, sometimes every 15 minutes. There's really not much we can do, with the exception of manually triggering the workflow using an external scheduler.

In #42, @belhyun proposed this:

> Thank you. In order to execute correctly every 5 minutes, I ran a build job with jenkins and executed a POST request.

Maybe that's the way to go if you want the workflow to run exactly every 5 minutes. Other cron scheduling services may also help you use the GitHub API to trigger a `workflow_dispatch` function to manually run Uptime CI. If you choose to do that, I recommend you change the schedule of the default "Uptime CI" to maybe once per year (`0 0 1 1 *`) so you don't have both services scheduling your workflow.

**Comment by AnandChowdhary** [comment]
I wrote a quick blog post with some additional information about this issue: https://upptime.js.org/blog/2021/01/22/github-actions-schedule-not-working.

**Reply by ririko5834** [reply]
Hello, can you please write tutorial how to setup IFTTT to run the workflow every 5 minutes using Github Api with IFTTT?

**Comment by PeterDaveHello** [comment]
Could use `repository_dispatch` event as a workaround (https://docs.github.com/en/rest/reference/repos#create-a-repository-dispatch-event), just use any other reliable environment to run the cron job, and trigger Uptime CI by it.

As GitHub Actions scheduler seems to be very unreliable, it's important to have a tutorial about how to deal with the workaround, because the monitor needs to be reliable, but now it relies on something not ...

**Reply by ririko5834** [reply]
I think that they should make guide how to setup IFTTT to make the workflow regular every 5 mins

**Reply by PeterDaveHello** [reply]
Yeah, just something like that but not must be IFTTT
